vividus-plugin-rest-api/src/main/java/org/vividus/http/CurlUtils.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.http;
+
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpRequest;
+
+public final class CurlUtils
+{
+    private static final String CURL_REQUEST_COMMAND = "curl -X %s '%s' %n";
+    private static final String HEADER = "-H '%s: %s' %n";
+    private static final String BODY = "-d '%s'";
+    private static final String FORM_TEXT = "-F \"%s=%s\" %n";
+    private static final String FORM_FILE = "-F \"%s=@<path-to-file>%s\" %n";
+
+    private static final Pattern NAME_FILE_NAME_PATTERN = Pattern.compile("name=\"(.+)\"; filename=\"(.+)\"");
+    private static final Pattern NAME_CONTENT_PATTERN = Pattern.compile("name=\"(.+)\"\nContent-Type:.+\n\n(.*)");
+
+    private CurlUtils()
+    {
+    }
+
+    public static String buildCurl(HttpRequest request, String mimeType, byte[] body)
+    {
+        return getRequestAsString(request)
+                + getHeadersAsString(request.getHeaders())
+                + getContentAsString(mimeType, body);
+    }
+
+    private static String getRequestAsString(HttpRequest request)
+    {
+        String requestString;
+        try
+        {
+            requestString = String.format(CURL_REQUEST_COMMAND, request.getMethod(), request.getUri());
+        }
+        catch (URISyntaxException e)
```

Comment:
it shouldn't be caught here, it make sense to throw it and catch on upper levels